framework/auth/core.py

```diff
@@ -888,15 +1065,24 @@ def merge_user(self, user, save=False):
                     user._id, node._id
                 ))
             node.save()
-        # Inherits projects the user created
+
+        # - projects where the user was the creator
         for node in user.node__created:
             node.creator = self
             node.save()
+
+        # finalize the merge
+
+        # - username is set to None so the resultant user can set it primary
+        #   in the future.
+        user.username = None
+        user.password = None
+        user.email_verifications = {}
+        user.verification_key = None
         user.merged_by = self
+
         user.save()
-        if save:
-            self.save()
-        return None
+        self.save()
```

Comment:
Done.

I've also made a note to consider alternatives to this, where side effects - particularly time-intensive side effects - might be deferred until the request is complete, regardless of when or how often `StoredObject.save()` is called.